src/nile/core/types/transaction.py

```diff
@@ -0,0 +1,343 @@
+"""Transaction module."""
+
+import dataclasses
+import json
+import logging
+import re
+from dataclasses import field
+from typing import List, Literal
+
+from starkware.cairo.common.hash_chain import compute_hash_chain
+from starkware.starknet.core.os.contract_address.contract_address import (
+    calculate_contract_address_from_hash,
+)
+from starkware.starknet.definitions.general_config import StarknetChainId
+from starkware.starknet.public.abi import get_selector_from_name
+
+from nile.common import TRANSACTION_VERSION, get_class_hash, get_contract_class, pt
+from nile.core.types.utils import (
+    get_declare_hash,
+    get_deploy_account_hash,
+    get_execute_calldata,
+    get_invoke_hash,
+)
+from nile.starknet_cli import execute_call
+from nile.utils import hex_address
+from nile.utils.status import status
+
+# Version for query calls
+QUERY_VERSION_BASE = 2**128
+
+
+@dataclasses.dataclass
+class Transaction:
```

Comment:
In the design I had in mind, the idea of the `Transaction` is to be low level. Users won't have to create a `Transaction` by themselves but use it implicit in the Nile flow, is more an improvement for the organization of code flow. 

In scripting they would get the transaction by calling "wrapper" methods like `Account.declare` or `Account.send`.

The purpose of`TxWrapper` is to handle logic specific to Nile, that is outside of a starknet `Transaction` flow: like registering or unregistering deployments in the deployments.txt file. Without this logic, we could return a `Transaction` object with the same interface from Account methods (it didn't seem right to add Nile logic to the `TRansaction` definition itself).

Reply:
users include plugin developers, which might interface with the Transaction class directly. it's expected for users to dive into these abstractions too. having different classes would also allow nile developers (us) to not mess.

i think it's a good practice to have representative abstractions and this Transaction class should not support all those attributes simultaneously.

with this proposal we can still have a separate TxWrapper class, it would just be simpler.

Reply:
Yes, that makes sense.